modules/core/src/main/scala/com.snowplowanalytics.snowplow.lakes/processing/SparkSchema.scala

```diff
@@ -22,11 +22,11 @@ object SparkSchema {
    *
    * The returned schema includes atomic fields and non-atomic fields but not the load_tstamp column
    */
-  private[processing] def forBatch(entities: Vector[TypedTabledEntity]): StructType = {
+  private[processing] def forBatch(entities: Vector[TypedTabledEntity], respectNullability: Boolean): StructType = {
     val nonAtomicFields = entities.flatMap { tte =>
       tte.mergedField :: tte.recoveries.map(_._2)
     }
-    StructType(atomic ++ nonAtomicFields.map(asSparkField))
+    StructType(atomic ++ nonAtomicFields.map(asSparkField(_, respectNullability)))
```

Comment:
Does it also affect top-level unstruct/contexts fields and not only nested fields?

If so, is it OK? I guess here we care only about nested fields?

Reply:
I think I've caused confusion about what I mean about a nested field.  If a schema has fields `abc` and `xyz`, then I'm referring to `unstruct_event_myvendor_myevent_1.abc` and `unstruct_event_myvendor_myevent_1.xyz` as nested fields.  Sure they are top level fields of the Iglu schema, but they are nested fields from the point of view of the events table.

Does that resolve your question, or have I just made it even more confusing?

Reply:
And I think I've cause confusion about top level fields 😁 

What I meant was not even `abc` field but rather whole parent `unstruct_event_myvendor_myevent_1` field (or `contexts_myvendor_myevent_1`), columns living next to atomic. These are also fields created by schema-ddl and they also have some nullability (both are nullable...?) It doesn't probably really matter.

Reply:
OK I see what you mean.

This whole PR was about taking columns which were previously non-nullable and making them nullable.  But the parent fields (e.g. `unstruct_event_myvendor_myevent_1`) were always nullable and always will be.  So no change is needed there.

Or the other way of answering.... relax, this works.  We tested it.